pymodbus/client/mixin.py

```diff
@@ -546,16 +546,16 @@ def convert_from_registers(
 
     @classmethod
     def convert_to_registers(
-        cls, value: Union[int, float, str], data_type: DATATYPE
+        cls, value: Union[float, str], data_type: DATATYPE
     ) -> List[int]:
         """Convert int/float/str to registers (16/32/64 bit).
 
         :param value: value to be converted:
-        :param data_type: data type to convert to
+        :param data_type: data type to be encoded as registers
         :returns: List of registers, can be used directly in e.g. write_registers()
         """
-        if data_type == cls.DATATYPE.STRING:
-            byte_list = value.encode()  # type: ignore[union-attr]
+        if isinstance(value, str):
```

Comment:
that is not the intention, and are wrong is the call is using value = "127" and DATATYPE.INT.

I want to check the datatype not the value.

Reply:
OK, `convert_to_registers("127", DATATYPE.INT)` is strange to me, but I have tried a different approach that permits this and satisfies `mypy`.  

Actually, the new code prevents a crash if you were to call `convert_to_registers(127, DATATYPE.STRING)`, I think.  Which is the purpose of tools like mypy :)